### PR TITLE
[6.x] Ds-602 Remove pattern template function

### PR DESCRIPTION
**Pull request by MarcinMr**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-602

## Summary

Pattern_template function usage was rewritten in the Bolt repository.

## Details

All the examples of the `pattern_template` function (for example pattern: 'image' )used in the Bolt repository (components and pattern lab) are rewritten.

## How to test

Pull the branch.  Search for examples of `pattern_template` usage, and confirm that there is no regression due to updating it.

## Release notes

Pattern_template function is deprecated.

A list of where `pattern_template()` function will be no longer in use in the future:

- buton-group (`@bolt-components-button-group/button-group.twig`)
- button (`@bolt-components-button/button.twig`)
- card (`@bolt-components-card-replacement/card-replacement.twig`)
- card-w-teaser (`@bolt-components-card-w-teaser/card-w-teaser.twig`)
- eyebrow (`@bolt-components-headline/eyebrow.twig`)
- flag (`@bolt/flag.twig`) (this is an object)
- headline (`@bolt-components-headline/headline.twig`)
- image (`@bolt-components-image/image.twig`)
- link (`@bolt-components-link/link.twig`)
- teaser (`@bolt-components-teaser/teaser.twig`)
- text (`@bolt-components-text/text.twig`)

`pattern` prop (`pattern: $name`) will no longer be used. Please use the bolt-components whenever you want to utilize card, headline, image, etc.

There is no need to use the `pattern: image` when utilizing `bolt-background` component.

### CODE REVIEW

**Review by colbytcook — CHANGES_REQUESTED** [review]
@MarcinMr there seems to be more examples that need to be updated here (thanks @remydenton for finding these). Here are an example of a few.
- https://github.com/boltdesignsystem/bolt/blob/feature/DS-602-Remove-pattern_template-Function/docs-site/src/pages/pattern-lab/_patterns/999-archive/product-pages/_product-t3-extra-videos.twig#L653
- https://github.com/boltdesignsystem/bolt/blob/feature/DS-602-Remove-pattern_template-Function/docs-site/src/pages/pattern-lab/_patterns/999-archive/product-pages/_product-t3-extra-videos.twig#L653 

There are examples of some other example being present on the some of the page that you fixed already.
- https://github.com/boltdesignsystem/bolt/blob/feature/DS-602-Remove-pattern_template-Function/docs-site/src/pages/pattern-lab/_patterns/40-components/breadcrumb/15-breadcrumb-use-case-band.twig#L151
- https://github.com/boltdesignsystem/bolt/blob/feature/DS-602-Remove-pattern_template-Function/docs-site/src/pages/pattern-lab/_patterns/40-components/breadcrumb/15-breadcrumb-use-case-band.twig#L240

Please make another pass here and see if there are more examples that need to be updated. Just for your reference you can pretty easily find any errors regarding this by running the `yarn build` command.

**Comment by MarcinMr** [comment]
@colbytcook @remydenton 

These examples you mentioned above have been corrected. `Yarn build` doesn't throw any errors on my side.

There were a lot of usage `pattern: 'image'` that I missed last time. I rewrite a little bit the `background-item` template to get rid of the `item.pattern = 'image'` statement and the all usage in the docs. I'm not sure if it's done correctly. I created a test page to see if it still works `/?p=tests-background-image`. It seems that for these two scenarios(passing image attributes as an object and pass the whole image as a variable) everything is fine. Let me know what do you think.

File: packages/components/bolt-background/src/background-item.twig [hunk 1]
```diff
@@ -2,66 +2,43 @@
...
-        background: true,
-        attributes: attributes|merge({
-          src: item.src,
-          sizes: _sizes,
-          loading: _loading,
-        })
-      } only %}
-    {% else %}
-      {% include pattern_template(item.pattern) with item|merge({
-        ratio: _ratio_value,
```
Comment (colbytcook) [inline packages/components/bolt-background/src/background-item.twig:60]: Should this ratio prop be remapped to a Ratio Element?

**Review by danielamorse — CHANGES_REQUESTED** [review]
@MarcinMr this will be a breaking change and we have to provide instructions for developers downstream on what props have been deprecated (still work but slated for removal) and what props have been removed (deleted and won't work).

There is a lot here, and a lot of very good cleanup, but I'm having trouble identifying the key changes that developers downstream will have to know about in order to pull in this code. Would you please add to the description of this PR exactly what components are affected and what props will change as a result of this PR. I'll use that in my review and we'll include those notes in the 6.x upgrade steps.

For example, here is how `contentItems` is being used in Drupal currently: https://gitlab.com/search?group_id=5804310&scope=blobs&search=%22contentItems%22

This may be a good opportunity to remove this prop completely, as long as we provide upgrade instructions to the Drupal team. Talk more in dev huddle.

cc @mikemai2awesome @remydenton

File: packages/components/bolt-background/src/background-item.twig [hunk 2]
```diff
@@ -2,66 +2,43 @@
...
 
-      {% set _sizes = item.sizes ? item.sizes : '100vw' %}
+    {% set _sizes = item.sizes ? item.sizes : '100vw' %}
 
-      {# Found both of these spellings, expanding the conditional to catch both #}
-      {% set _loading = (item.lazyLoad is same as false) or (item.lazyload is same as false) ? 'eager' : 'lazy' %}
+    {# Found both of these spellings, expanding the conditional to catch both #}
+    {% set _loading = (item.lazyLoad is same as false) or (item.lazyload is same as false) ? 'eager' : 'lazy' %}
 
-      {% include '@bolt-elements-image/image.twig' with {
```
[the remark this answers is not in the document]
Comment (remydenton) [reply packages/components/bolt-background/src/background-item.twig:50]: The following section of this file actually allows you to pass items beside images:
```
{% elseif (item is iterable) %}
  <div class="{{ 'c-bolt-background__item' }}">
    {{ item }}
  </div>
{% else %}
  {{ item }}
{% endif %}
```
Why the `is iterable` condition?  it's not clear.  It was added in 8ff7e364c, which claims to be just breaking one template into smaller ones.  In any case, it's poor practice to use `is iterable` like this, and I would say the whole file should just look like the following after we're done removing `pattern_template()`:

```
{% if item %}
  <div class="{{ 'c-bolt-background__item' }}">
    {{ item }}
   </div>
{% endif %}
```

Of course, we'll want to leave the existing logic in the PR that is backwards-compatible (for the 5.x branch).  But in both PRs, we should update all pattern lab examples to pass data the correct/new way, that is, like this:
```
{% include '@bolt-elements-image/image.twig' with {
    background: true,
    attributes: {
      alt: 'Alt text.',
      loading: 'lazy',
      src: '/images/content/backgrounds/background-tall-4.jpg',
      width: 3840,
      height: 2160,
    },
  } only %}
{% endset %}
{% include '@bolt-components-background/background.twig' with {
  opacity: 'heavy',
  fill: 'gradient',
  content_items: [
    image
  ]
} only %}
```
:white_check_mark:

i.e. remove/replace all examples that look like this:
```
{% include '@bolt-components-background/background.twig' with {
    opacity: 'heavy',
    fill: 'gradient',
    content_items: [
      {
        alt: 'Alt text.',
        loading: 'lazy',
        src: '/images/content/backgrounds/background-tall-4.jpg',
        width: 3840,
        height: 2160,
      },
    ]
} only %}
```
:x:

[the remark this answers is not in the document]
Comment (MarcinMr) [reply packages/components/bolt-background/src/background-item.twig:50]: Updated examples in pattern lab to use the new way Remy mentioned above

**Comment by colbytcook** [comment]
https://gitlab.com/pegadigital/bolt/design-system/-/merge_requests/3

**Pull request closed without merging**